### PR TITLE
#7 Huy's project 

### PR DESCRIPTION
**Pull request by huy525**

### CODE REVIEW

**Comment by Ddorsmt** [comment]
I _think_ mint and burn functions are specified to be required.
Also mint _should_ be executed by owner only
No issues with tests.

**Comment by PendlingMatt** [comment]
- Burn function seems to be missing. 
- Can make use of "require" instead of writing "if" blocks for function call validation.
- Not sure if `_allowance` needs to be decreased after `transferFrom` is called

Very comprehensive tests! 👍 

Edit: Realised that `_allowance` should indeed be decremented after `transferFrom`, my bad!

**Comment by huy525** [comment]
Thanks for your feedbacks! I've added burn() and changed "if"s to "require"s, but I'll keep the mint/burn functions public for now.

File: contracts/ERC20.sol [hunk 1]
```diff
@@ -1,26 +1,54 @@
...
 
-import "./IERC20.sol";
+import "./IERC20Metadata.sol";
+
+contract ERC20 is IERC20Metadata {
+  string internal _name;
+  string internal _symbol;
+  uint8 internal _decimals;
+  uint256 internal _totalSupply;
+  mapping(address => uint256) internal _balanceOf;
```
Comment (UncleGrandpa925) [inline contracts/ERC20.sol:11]: good code overall!

**Pull request closed without merging**